File: solution/713. Subarray Product Less Than K/713.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        int ans = 0;
        
        int i = 0, prod = 1;
        for (int j = 0; j < nums.size(); ++j) {
            prod *= nums[j];
            while (i <= j && prod >= k) {
                prod /= nums[i++];
            }
            ans += j - i + 1;
        }
        return ans;
    }
};

File: solution/713. Subarray Product Less Than K/713.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        res = 0
        prod, i = 1, 0
        for j in range(len(nums)):
            prod *= nums[j]
            while i <= j and prod >= k:
                prod = prod // nums[i]
                i += 1
            res += j - i + 1
        return res
    
